src/sql-parser/src/parser.rs

```diff
@@ -2982,8 +2982,8 @@ impl<'a> Parser<'a> {
     }
 
     /// Parse a SQL datatype (in the context of a CREATE TABLE statement for example)
-    fn parse_data_type(&mut self) -> Result<DataType<Raw>, ParserError> {
-        let other = |name: &str| DataType::Other {
+    fn parse_data_type(&mut self) -> Result<UnresolvedDataType, ParserError> {
```

Comment:
Looks like we're missing the ability to parse resolved data types on reboot, i.e. this is what gets serialized in the catalog:

```
"CREATE TABLE \"materialize\".\"public\".\"t\" (\"a\" [s1003 AS \"int4\"])
```

Reply:
Sigh, thanks. Fixed, I think. That _is_ what we want to serialize in the catalog; just the parser wasn't handling it.